Library/Homebrew/cask/lib/hbc/cask_loader.rb

```diff
@@ -31,6 +31,12 @@ def initialize(path)
 
         @token = path.basename(".rb").to_s
         @path = path
+
+        @tap ||= if @path == CaskLoader.default_path(@token)
```

Comment:
I tried out changing `Hbc::Scopes::installed` (used by the `brew cask list` command) to fetch casks by tap path, thus using `FromTapLoader` (see the [patch](https://gist.github.com/wendorf/10619660bcba7c8450f91bce7df32809)), but failed the `rspec ./test/cask/scopes_spec.rb:21 # Hbc::Scopes installed optimizes performance by resolving to a fully qualified path before calling Hbc::CaskLoader.load` test. It looks like this optimization was put there for good reason, so I'm hesitant to change that behavior.

@reitermarkus Is there another option you think I should consider? Or do you think my existing approach is okay since the path checking is checking explicitly for the Homebrew tap install locations? I think path checking is the only way to do it without making seemingly-undesirable changes to how `brew cask list` gets its casks.